src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -218,7 +219,31 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                             targetsList = await GetTargetsListAsync(assetsFilePath, token);
                         }
 
-                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, token) : new Dictionary<string, TransitiveEntry>();
+                        Dictionary<string, TransitiveEntry> transitiveOriginsCache = new();
+
+                        // Compute Transitive Origins
+                        transitiveOrigins = calculatedTransitivePackages.Any() ? ComputeTransitivePackageOrigins(calculatedInstalledPackages, targetsList, transitiveOriginsCache, token) : transitiveOriginsCache;
+
+                        // If the project has project references, we need to compute transitive origins for their packages
+                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
+                        if (projectReferences.Any())
```

Comment:
I am curious to know what the user experience will be if the referenced project has another project reference which has a package reference.

Project A -> Project B -> Project C (has a package reference to Newtonsoft.Json)

Will the PM UI display `Newtonsoft.Json` as an installed package for `Project A`? I assume it will not, but I wanted to understand better because this PR is intended to display all the transitive packages in the PM UI for a project.

Reply:
If you open the PM UI for Project A/B it will display `Newtonsoft.Json` as a **transitive** package. That's the only thing that will be in the PM UI packages list.

If you open the PM UI for Project C it will display `Newtonsoft.Json` as a top level package, so the current behavior.